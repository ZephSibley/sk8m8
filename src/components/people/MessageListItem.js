import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ListItem from '@material-ui/core/ListItem';
import Divider from '@material-ui/core/Divider';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles(theme => ({
    root: {
      width: '100%',
      maxWidth: 360,
      backgroundColor: theme.palette.background.paper,
    },
    inline: {
      display: 'inline',
    },
}));


const MessageListItem = props => {
    // Prop: sender; string
    // Prop: avatar; string, url src
    // Prop: message; string
    const classes = useStyles();

    return (
        <React.Fragment>
            <ListItem alignItems="flex-start">
                <ListItemAvatar>
                <Avatar
                    alt={props.sender}
                    src={props.avatar}
                    //imgProps={loading='lazy'}
                />
                </ListItemAvatar>
                <ListItemText
                primary={props.sender}
                secondary={
                    <Typography
                        component="span"
                        variant="body2"
                        className={classes.inline}
                        color="textPrimary"
                    >
                        {props.message}
                    </Typography>
                }
                />
            </ListItem>
            <Divider variant="inset" component="li" />
        </React.Fragment>
    )
}

export default MessageListItem;