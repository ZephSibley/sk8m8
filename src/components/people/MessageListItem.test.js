import React from 'react';
import renderer from 'react-test-renderer';
import MessageListItem from './MessageListItem';

test('Basic Rendering', () => {
    const component = renderer.create(
        <MessageListItem
            sender='Zeph'
            avatar=''
            message='Hi There!'
        />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})