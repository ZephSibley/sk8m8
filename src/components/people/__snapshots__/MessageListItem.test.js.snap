// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
    disabled={false}
  >
    <div
      className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
    >
      <div
        className="MuiAvatar-root MuiAvatar-colorDefault"
      />
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Zeph
      </span>
      <span
        className="MuiTypography-root makeStyles-inline-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
      >
        Hi There!
      </span>
    </div>
  </li>,
  <li
    className="MuiDivider-root MuiDivider-inset"
    role="separator"
  />,
]
`;
